test-data/unit/check-typeddict.test

```diff
@@ -2012,6 +2012,155 @@ v = {bad2: 2}  # E: Extra key "bad" for TypedDict "Value"
 [builtins fixtures/dict.pyi]
 [typing fixtures/typing-typeddict.pyi]
 
+[case testOperatorContainsNarrowsTypedDicts_unionWithList]
+from __future__ import annotations
+from typing import assert_type, TypedDict, Union
+from typing_extensions import final
+
+@final
+class D(TypedDict):
+    foo: int
+
+
+d: D | list[str]
+
+if 'foo' in d:
+    assert_type(d, Union[D, list[str]])
+else:
+    assert_type(d, list[str])
+
+[builtins fixtures/dict.pyi]
+[typing fixtures/typing-typeddict.pyi]
+
+[case testOperatorContainsNarrowsTypedDicts_total]
+from __future__ import annotations
+from typing import assert_type, Literal, TypedDict, TypeVar, Union
+from typing_extensions import final
+
+@final
+class D1(TypedDict):
+    foo: int
+
+
+@final
+class D2(TypedDict):
+    bar: int
+
+d: D1 | D2
+opt_d: D1 | None
+
+if 'foo' in opt_d:
```

Comment:
This should be a type error. `'foo' in None` raises.

Reply:
Requires strict optional. I can enable it, but it's not what we're testing here.

Reply:
We want to enable `--strict-optional` by default sometime soon, so it will fail in the currect setup.

You can just change `D1 | None` to be `D1 | list[str]` or something similar.

Reply:
a47bc04e0f1f916437df4b6cbcb00d35a422a6de